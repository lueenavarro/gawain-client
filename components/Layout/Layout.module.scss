.layout {
  &__main {
    height: 100vh;
  }
}
