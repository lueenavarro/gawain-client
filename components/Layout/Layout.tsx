import Head from "next/head";
import React from "react";

import style from "./Layout.module.scss";

const Layout = ({ children }) => {
  return (
    <>
      <Head>
        <title>Gawain</title>
        <meta charSet="utf-8" />
        <meta name="viewport" content="initial-scale=1.0, width=device-width" />
        <meta name="description" content="Do your thing one step at a time" />
      </Head>
      <main className={style["layout__main"]} data-testid="main">{children}</main>
    </>
  );
};

export default Layout;
