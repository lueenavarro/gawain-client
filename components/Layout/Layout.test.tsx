import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

import Layout from "./Layout";

test("render without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Layout><div>Hello world!</div></Layout>, div);
})

test("snapshot matches", ()=> {
    const tree= renderer.create(<Layout><p>Hello world!</p><p>Hi There!</p></Layout>).toJSON();
    expect(tree).toMatchSnapshot();
})