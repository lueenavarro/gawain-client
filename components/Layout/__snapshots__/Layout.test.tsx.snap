// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot matches 1`] = `
<main
  className="layout__main"
  data-testid="main"
>
  <p>
    Hello world!
  </p>
  <p>
    Hi There!
  </p>
</main>
`;
